src/main/java/com/facebook/openwifirrm/modules/ModelerUtils.java

```diff
@@ -381,4 +399,269 @@ public static Map<String, Map<String, WifiScanEntry>> getAggregatedWifiScans(
+			break;
+		default:
+			serializer =
+				new JsonSerializer<MeanAggregator>() {
+					/**
+					 * If the aggregation type is to take mean of the values,
+					 * return a JsonPrimitive type of single value string.
+					 */
+					@Override
+					public JsonElement serialize(
+						MeanAggregator agg,
+						java.lang.reflect.Type typeOfSrc,
+						JsonSerializationContext context
+					) {
+						return new JsonPrimitive(agg.getAggregate().toString());
+					}
+				};
+			break;
+		}
+		gsonBuilder.registerTypeAdapter(AggregationType, serializer);
+		Gson customGson = gsonBuilder.create();
+		String aggregatedJson = customGson.toJson(aggregatedMap);
+		return aggregatedJson;
+	}
+
+	/** This method reports the agggregated statistics as a Json String.
+	 *
+	 * @see #getAggregatedStatesJson(DataModel, long, List, List, AggregationType)
+	 *
+	 * @param dataModel 			the data model which includes the latest recorded States.
+	 * @param obsoletionPeriodMs	the maximum amount of time (in milliseconds) it
+	 *                           	is worth aggregating over.
+	 * @param aggType				Determines how to aggregate the statistics.
+	 * @return String				A Json String of aggregation statistics.
+	 */
+	public static String getAggregatedStatesJson(
+		Modeler.DataModel dataModel,
+		long obsoletionPeriodMs,
+		AggregationType aggType
+	) {
+		return getAggregatedStatesJson(
+			dataModel,
+			obsoletionPeriodMs,
+			new ArrayList<>(
+				Arrays.asList("channel", "channel_width", "tx_power")
+			),
+			new ArrayList<>(Arrays.asList("rssi")),
+			aggType
+		);
+	}
+
+	/** This method serializes the aggregated statistics as a Json String.
+	 * How to aggregate the stats (mean/list) is defined by the input parameter.
+	 *
+	 * @see #getAggregatedStates(DataModel, long, long, List, List)
+	 *
+	 * @param dataModel 			the data model which includes the latest recorded States.
+	 * @param obsoletionPeriodMs	the maximum amount of time (in milliseconds) it
+	 *                           	is worth aggregating over
+	 * @param aggregatedKeys 		In which conditions the stats results need to be aggregated.
+	 * @param aggregatedFields		Which fields to be aggregated. RSSI, MCS for example.
+	 * @param aggType				Determines how to aggregate the statistics.
+	 * @return String				A Json String of aggregation statistics.
+	 */
+	public static String getAggregatedStatesJson(
+		Modeler.DataModel dataModel,
+		long obsoletionPeriodMs,
+		List<String> aggregatedKeys,
+		List<String> aggregatedFields,
+		AggregationType aggType
+	) {
+		Map<String, Map<String, Map<String, Map<JsonNode, Map<String, MeanAggregator>>>>> aggregatedMap =
+			new HashMap<>();
+		try {
+			aggregatedMap = getAggregatedStates(
+				dataModel,
+				obsoletionPeriodMs,
+				System.currentTimeMillis(),
+				aggregatedKeys,
+				aggregatedFields
+			);
+		} catch (
+			IllegalArgumentException | IllegalAccessException
+			| NoSuchFieldException | SecurityException e
+		) {
+			e.printStackTrace();
```

Comment:
`e.printStackTrace()` should almost never be present in production code - use `logger.error("description", e)`